dataland-frontend/src/components/forms/Types.ts

```diff
@@ -0,0 +1,16 @@
+import { CompanyReport } from "@clients/backend";
+
+export interface ExtendedFile extends File {
+  reportDate: string;
+  convertedReportDate: string | Date;
+  documentId: string;
+  [key: string]: unknown;
+}
+
+export interface ExtendedCompanyReport extends CompanyReport {
+  name: string;
+  convertedReportDate: string | Date;
+  [key: string]: unknown;
+}
+
+export type WhichSetOfFiles = "uploadFiles" | "filesToUpload";
```

Comment:
This is a little confusing for me as someone who has not participated in this story.

Why do we need those types?
Why do both types have a redundant info in "convertedReportDate"?
Why do they need to be able to accept more fields?

Reply:
We need them because I need to add auxiliary fields to the object describing the selected file. So to know which has already been uploaded, which has not, which already exists on the server and which has not.

convertedReportDate we need it because we convert the data type and format. We send date in format: yyyy-mm-dd and we count it from the Date object

The [key: string]: unknown entry is to satisfy the typescript so that it doesn't throw an error when we searching by string type in an object.

Reply:
Note: Florian and I will investigate if we can simplify this a little bit.